itest/lnd_multi-hop_test.go

```diff
@@ -1838,3 +1838,165 @@ func createThreeHopNetwork(ht *lntest.HarnessTest,
 
 	return aliceChanPoint, bobChanPoint, carol
 }
+
+// testExtraMempoolPreimage tests that in the multi-hop setting,
+// Alice->Bob->Carol, if the outgoing HTLC is about to time out, Alice will go
+// on chain to claim it. Meanwhile, Bob decides to spend the HTLC on-chain
+// using Alice's commitment transaction via the preimage path. In this
+// scenario, the node that sent the outgoing HTLC(Alice) should extract the
+// preimage from the sweep transaction as soon as the transaction appears in
+// mempool, and finish settling the HTLC backwards into the route. In addition,
+// Alice should also try to claim the on-chain HTLC by fee bumping the 2nd
+// level timeout transaction.
+//
+// TODO(yy): update the test to check fee bumping is in effect.
+func testExtractMempoolPreimage(ht *lntest.HarnessTest) {
+	runMultiHopHtlcClaimTest(ht, runExtraPreimageFromMempool)
+}
+
+func runExtraPreimageFromMempool(ht *lntest.HarnessTest,
+	alice, bob *node.HarnessNode, c lnrpc.CommitmentType, zeroConf bool) {
+
```

Comment:
Needs to be skipped for neutrino. Also seems to have a flake according to the travis build?